src/journal/journald-server.c

```diff
@@ -2051,10 +2052,12 @@ int server_init(Server *s) {
         if (r < 0)
                 return r;
 
-        /* /dev/ksmg */
-        r = server_open_dev_kmsg(s);
-        if (r < 0)
-                return r;
+        /* /dev/kmsg */
+        if (s->read_kmsg) {
+                r = server_open_dev_kmsg(s);
```

Comment:
yes, the fd is opened O_RDWR and we use it both for forwarding and for reading. Moreover any such option tests should really move into server_open_dev_kmsg() itself, i.e. in the callee, not in the caller